### PR TITLE
Add chat2-matterbridge

### PR DESCRIPTION
**Pull request by jm-clius**
This is an incremental PR towards https://github.com/status-im/nim-waku/issues/404.

## Background

It introduces a bridge between `chat2`  and a configurable [`matterbridge`](https://github.com/42wim/matterbridge) instance. Matterbridge can then be separately configured to support bridging from `chat2` (over Waku v2) to a large number of protocols, including Discord.

This basic implementation includes:
1. A Matterbridge Client utility class that interfaces with a Matterbridge API. It provides the ability to `post` new messages, `poll` for new messages and perform health checks
2. A bridge between `chat2` and Matterbridge that translates between `chat2` messages and Matterbridge
3. A chat2-Matterbridge configuration file

I've tested this locally and can confirm that, with a correctly configured Matterbridge client, the `chat2bridge` can successfully bridge between `chat2` and Discord.

## Next steps
1. Duplicate filtering
1. Re-use embedded timestamps from messages in both directions rather than using system time
1. Add Matterbridge config file template to illustrate how to set it up for bridging to Discord
1. Deploy a `chat2bridge` to the status fleet and connect to `test` cluster

## Possible improvements
- [ ] Better error handling
- [ ] Perform regular health checks
- [ ] Consider adding security token to matterbridge config
- [ ] Improved metrics and logs

### CODE REVIEW

**Comment by EbubeUd** [comment]
I get an OSError when I try to run chat2bridge locally.

**Comment by staheri14** [comment]
Is there any instructions on how to run `chat2bridge`? @jm-clius

File: waku/common/README.md [hunk 1]
```diff
@@ -5,3 +5,4 @@ This folder contains (a) modules that use both Waku v1 and Waku v2. and (b) util
 Examples include:
 - Bridge between v1 and v2
 - NAT traversal
+- interworking with protocols external to Waku
```
Comment (oskarth) [inline waku/common/README.md:8]:
```suggestion
- interworking with protocols external to Waku (such as Matterbridge)
```

**Review by oskarth — APPROVED** [review]
LGTM

**Comment by jm-clius** [comment]
> I get an OSError when I try to run chat2bridge locally.

Thanks! Have you provided it with a local Matterbridge instance? That said, error handling can certainly be improved. I will also add a tutorial soon.

**Comment by jm-clius** [comment]
> Is there any instructions on how to run `chat2bridge`? @jm-clius

Not yet, but tutorial coming soon.

**Comment by EbubeUd** [comment]
> > I get an OSError when I try to run chat2bridge locally.
> 
> Thanks! Have you provided it with a local Matterbridge instance? That said, error handling can certainly be improved. I will also add a tutorial soon.

Ohh... Not yet. I guess that must have been the problem. Well, It'll be fairly straightforward when the guidelines are available. 
Asides that, LGTM

**Review by EbubeUd — APPROVED** [review]

**Pull request merged**